### DISCUSSION TITLE
Tests fail to build in Xcode 16 beta

### DISCUSSION THREAD

**Discussion by bdbergeron** [pointfreeco/swift-case-paths#169] [Q&A]
**Describe the bug**
I'm using this library as part of a feature that uses TCA, and I believe that it's the CasePaths library specifically that's causing my test suite to fail to build. The app itself builds and runs as expected.

**To Reproduce**
 I have two tests that cover both the success and failure cases of an `Action` that has an associated value of `Result<Void, Error>` type, used within a feature annotated with the `@Reducer` macro:

```swift
@Reducer
struct MyFeature {
  enum Action: Sendable {
    case didFinishLoading(Result<Void, Error>)
    ...
  }
  ...
}
```

My test cases look like this:

```swift
final class MyFeatureTests: XCTestCase {
  @MainActor
  func test_fetch_success() async {
    let store = TestStore(initialState: MyFeature.State()) {
        MyFeature()
    }
    await store.send(.fetch) {
      $0.isFetching = true
    }
    await store.receive(\.setFetchingFinished.success) {
      $0.isFetching = false
    }
  }

  @MainActor
  func test_fetch_failure() async {
    let store = TestStore(initialState: MyFeature.State()) {
      MyFeature()
    } withDependencies: {
      $0.fetchAction = {
        throw TestError.expected
      }
    }
    await store.send(.fetch) {
      $0.isFetching = true
    }
    await store.receive(\.setFetchingFinished.failure) {
      $0.isFetching = false
      $0.navigationDestination = .alert(AlertState { TextState("Expected error.") })
    }
  }
}

private enum TestError: LocalizedError {
  case expected

  var errorDescription: String? {
    switch self {
    case .expected:
      "Expected error."
    }
  }
}
```

**Expected behavior**
The test suite should build without error.

**Environment**
  - swift-case-paths 1.4.2
  - Xcode 16.0 beta 1
  - Swift 6.0
  - OS: macOS Sonoma 14.5

**Additional context**
Changing the `Void` success type to some other concrete type such as Bool results in the tests building successfully. Also removing the `.success` and `.failure` from the case path in the `.receive()` calls results in the tests building successfully.

I use the `Void` type in this instance because I have an external (non-TCA) object that handles the loading and data management that the `.fetch` action triggers.

**Comment by stephencelis** [comment]
@bdbergeron Can you zip up and share a project that reproduces the issue? The code snippet above unfortunately isn't enough to debug things.

**Comment by bdbergeron** [comment]
@stephencelis sure thing, here y'a go: [TCATestIssueRepro.zip](https://github.com/user-attachments/files/15830932/TCATestIssueRepro.zip)

**Reply by Alex293** [reply]
@bdbergeron did you report this to Apple/Swift ?

**Reply by bdbergeron** [reply]
@Alex293 Unfortunately I have not had time to do so in the past many days, having to focus on another project at the moment.

**Comment by stephencelis** [comment]
@bdbergeron As far as I can tell this is a Swift bug/regression that affects all the way back to swift-case-paths 1.1, where the macros were first introduced, and I haven't been able to figure out a workaround in the library itself to support the preexisting types.

Can you file a bug with apple/swift and report back here? I'll try to spend some time reducing the problem tomorrow, but it'd be good to flag the regression with Apple.

Since this is an Apple bug that we can't do much about, I'm going to convert to a discussion, but hopefully it gets fixed soon.

**Comment by agruchala** [comment]
Still present on official Xcode 16 release...

**Reply by leobouilloux** [reply]
Same issue on our side 😢

**Reply by mbrandonw** [reply]
Unfortunately I don't think there is much we can do to mitigate this. It might be better to migrate to `Error?` than `Result<Void, Error>`. Another option is to define your own void type to work around:

```swift
struct Empty {}
case response(Result<Empty, Error>)
```

That seems to work for me.

**Reply by leobouilloux** [reply]
Thank you for your feedback, we will try this suggestion

**Comment by stephencelis** [comment]
For those that would like to follow, I've reduced and filed the bug with Apple here: https://github.com/swiftlang/swift/issues/76607
